Dockerfile

```diff
@@ -1,13 +1,28 @@
 # base image
-FROM node:12.2.0-alpine
+# Vite did not like the older version of Node that we'd specified, so I switched to the most recent LTS version
+FROM node:hydrogen-alpine
+
 
 # set working directory
 WORKDIR /app
 
 # install and cache app dependencies
 COPY package.json /app/package.json
 RUN npm install
-RUN npm install @vue/cli@3.7.0 -g
 
-# # start app
-CMD ["npm", "run", "serve"]
\ No newline at end of file
+# Without this line the page will not load
+COPY . . 
+
+EXPOSE 8082
+
+# Without this section, I was getting continual errors
+# when running docker-compose.
+# I found the solution here: https://www.reddit.com/r/docker/comments/ww6rt5/eacces_permission_denied/
+RUN mkdir /app/node_modules/.vite \
+    && chown -R node:node /app/node_modules/.vite
+
+USER node
```

Comment:
I think the issue you described in your comment [here](https://github.com/wikimedia/toolhunt-ui/pull/5/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R18-R20) is due to this line. The **node** user doesn't seem to have root privileges in this container, hence this may cause some additional issues in the future and it wouldn't be optimal to add every directory to our DockerFile. 

I think we can fix this by either removing this line or using USER root instead of adding additional READ rights to the file as you did on line 21.

Let me know if this works.

Reply:
A better approach  (which I believe is where you were heading) to this would be one where we follow the [Principle of Least Privilege](https://www.infoq.com/articles/securing-docker/#:~:text=The%20principle%20of%20the%20least%20privilege%20says%20we%20should%20give,container%20with%20the%20root%20user.) would be to give the node user read privilege over everything within the container. Like this:

```
# base image
# Vite did not like the older version of Node that we'd specified, so I switched to the most recent LTS version
FROM node:hydrogen-alpine


# set working directory
WORKDIR /app

# Give node user READ privilege in the app directory
RUN chown -R node: /app

# install and cache app dependencies
COPY package.json /app/package.json
RUN npm install

EXPOSE 8082

USER node

# # start app 
# "dev" rather than "serv" is Vite's default for loading the development environment
CMD ["npm", "run", "dev"]
```

Reply:
Thank you for the insight, Damilare.

Reply:
Fixed with [3bfc29c](https://github.com/wikimedia/toolhunt-ui/pull/5/commits/3bfc29c49afaf1ac8b12ccef452c1c211d14249e)